ui/hooks/useFundPendingCampaign.ts

```diff
@@ -0,0 +1,110 @@
+import { toAscii, toBase64 } from "@cosmjs/encoding"
+import { findAttribute } from "@cosmjs/stargate/build/logs"
+import { useCallback, useState } from "react"
+import { useRecoilValue, useSetRecoilState } from "recoil"
+
+import { defaultExecuteFee } from "../helpers/config"
+import { prettyPrintDecimal } from "../helpers/number"
+import { globalLoadingAtom } from "../state/loading"
+import { signedCosmWasmClient } from "../state/web3"
+import { useRefreshCampaign } from "./useRefreshCampaign"
+import { useWallet } from "./useWallet"
+
+export const useFundPendingCampaign = (campaign: Campaign | null) => {
+  const client = useRecoilValue(signedCosmWasmClient)
+  const { walletAddress } = useWallet()
+
+  const setLoading = useSetRecoilState(globalLoadingAtom)
+  const { refreshCampaign } = useRefreshCampaign(campaign)
+  const [fundPendingCampaignError, setFundPendingCampaignError] = useState(
+    null as string | null
+  )
+
+  // TODO: Attempt to connect manually before so we can show connection errors.
+  const fundPendingCampaign = useCallback(
+    async (amount: number) => {
+      setFundPendingCampaignError(null)
+
+      if (!client) {
+        setFundPendingCampaignError("Failed to get signing client.")
+        return false
+      }
+      if (!walletAddress) {
+        setFundPendingCampaignError("Wallet not connected.")
+        return false
+      }
+      if (!campaign) {
+        setFundPendingCampaignError("Campaign is not loaded.")
+        return false
+      }
+
+      setLoading(true)
+      const cosmMsg = {
+        wasm: {
+          execute: {
+            contract_addr: campaign.dao.govToken.address,
+            msg: toBase64(
+              toAscii(
+                JSON.stringify({
+                  send: {
+                    contract: campaign.address,
+                    amount: (amount * 1e6).toString(),
+                    msg: "",
+                  },
+                })
+              )
+            ),
+            funds: [],
+          },
+        },
+      }
+
+      try {
+        const msg = {
+          propose: {
+            title: `Activate DAO Up! campaign`,
+            description: `Send ${prettyPrintDecimal(amount)} ${
+              campaign.dao.govToken.symbol
+            } to ${campaign.name} campaign on DAO Up! in order to launch it.`,
```

Comment:
We support markdown as of yesterday so let's link to the campaign :)

```suggestion
            } to [${campaign.name}](https://daoup.zone/campaign/${campaign.address}) campaign on DAO Up! in order to launch it.`,
```

Reply:
No way! Sweet